### DISCUSSION TITLE
Query template Date comparison with today and Time zone

### DISCUSSION THREAD

**Discussion by ghost** [microsoft-search/pnp-modern-search#1697] [Q&A]
In the Search Results web part (v4), what is the correct way to compare dates with Today?

I'm using SharePoint Online. SharePoint Search is the source and my Query Template contains 'NotificationDate>=today'
I noticed however that this comparison does not take into account the time zone:
- NotificationDate is a custom column in the Sites Pages library (Date and Time field set to Date Only)
- NotificationDate is stored as UTC by SharePoint
- As I'm not in the UTC timeszone, the site is configured with a different time zone 

So the comparison 'NotificationDate>=today' is comparing NotificationDate in UTC time with Today in my actual time zone. This results in unwanted behavior where some posts do not show up when they should.

How do I tell the Query Template in the Search Results web part to compare NotificationDate and Today in the same time zone (either UTC or mine)?

Thank you!

**Comment by wobba** [comment]
Can you try to use the `{Today}` query variable instead of the `Today` KQL keyword as we replace that client side with your local date and time translated to UTC. https://github.com/microsoft-search/pnp-modern-search/blob/main/search-parts/src/services/tokenService/TokenService.ts#L426

**Reply by ghost** [reply]
Thank you for your suggestion. Unfortunately, this does not work.
While both Today and {Today} are replaced by the correct time zone, NotificationDate is not.

So when is set NotificationDate to todays date January 11, it is stored by SharePoint in UTC format as 2022-01-**10**T23:00:00Z.
But when using NotificationDate is the Query template field, it is not converted back to the date I entered which should be 2022-01-11T00:00:00Z

So either Today should be in UTC format, or NotificationDate should be in local date/time for the comparison to work correctly.

**Reply by wobba** [reply]
SharePoint adjusts the time according to the time zone settings for the site. Is "2022-01-10T23:00:00Z" correct for what you expect when you set Jan 11? Or do you expect "2022-01-11T00:00:00Z"?

`{Today}` as a query variable if you inspect the JSON request should be your current time in UTC including hours, minutes and seconds.

If you need to limit differently you first need to get a handle on how data is stored, and then think about how time zones should or should not impact what you enter in a query.
